src/vs/workbench/contrib/chat/browser/chatDragAndDrop.ts

```diff
@@ -320,55 +323,54 @@ export class ChatDragAndDrop extends Themable {
 		});
 	}
 
-	// private async resolveHTMLAttachContext(data: string): Promise<IChatRequestVariableEntry[]> {
-	// 	const displayName = localize('dragAndDroppedImageName', 'Image from URL');
-	// 	let finalDisplayName = displayName;
-
-	// 	for (let appendValue = 2; this.attachmentModel.attachments.some(attachment => attachment.name === finalDisplayName); appendValue++) {
-	// 		finalDisplayName = `${displayName} ${appendValue}`;
-	// 	}
-
-	// 	const { src, alt } = extractImageAttributes(data);
-	// 	finalDisplayName = alt ?? finalDisplayName;
-
-	// 	if (/^data:image\/[a-z]+;base64,/.test(src)) {
-	// 		const resizedImage = await resizeImage(src);
-	// 		return [{
-	// 			id: await imageToHash(resizedImage),
-	// 			name: finalDisplayName,
-	// 			value: resizedImage,
-	// 			isImage: true,
-	// 			isFile: false,
-	// 			isDirectory: false
-	// 		}];
-	// 	} else if (/^https?:\/\/.+/.test(src)) {
-	// 		const url = new URL(src);
-	// 		const isImage = /\.(jpg|jpeg|png|gif|webp)$/i.test(url.pathname);
-	// 		if (isImage) {
-	// 			const buffer = convertStringToUInt8Array(src);
-	// 			return [{
-	// 				kind: 'image',
-	// 				id: url.toString(),
-	// 				name: finalDisplayName,
-	// 				value: buffer,
-	// 				isImage,
-	// 				isFile: false,
-	// 				isDirectory: false,
-	// 				isURL: true,
-	// 			}];
-	// 		} else {
-	// 			return [{
-	// 				kind: 'link',
-	// 				id: url.toString(),
-	// 				name: finalDisplayName,
-	// 				value: URI.parse(url.toString()),
-	// 				isFile: false,
-	// 				isDirectory: false,
-	// 			}];
-	// 		}
-	// 	}
-	// 	return [];
-	// }
+	private async downloadImageAsUint8Array(url: string): Promise<Uint8Array | undefined> {
+		try {
+			const extractedImages = await this.webContentExtractorService.readImage(URI.parse(url));
+			if (extractedImages) {
+				return extractedImages.buffer;
+			}
+		} catch (error) {
+			console.warn('Fetch failed:', error);
+		}
+
+		this.notificationService.error(localize('fetchFailed', 'Failed to fetch image from URL: {0}', url));
+		return undefined;
+	}
+
+	private async resolveHTMLAttachContext(e: DragEvent): Promise<IChatRequestVariableEntry[]> {
+		const displayName = localize('dragAndDroppedImageName', 'Image from URL');
+		let finalDisplayName = displayName;
+
+		for (let appendValue = 2; this.attachmentModel.attachments.some(attachment => attachment.name === finalDisplayName); appendValue++) {
+			finalDisplayName = `${displayName} ${appendValue}`;
+		}
+
+		const src = await extractImageAttributes(e);
+		const isString = typeof src === 'string';
+
+		if ((src instanceof Uint8Array && src.length > 0) || (isString && /^data:image\/[a-z]+;base64,/.test(src))) {
+			return [await this.createImageVariable(await resizeImage(src), finalDisplayName)];
+		} else if (isString && /^https?:\/\/.+/.test(src)) {
```

Comment:
For `text/uri-list`, you should use the `UriList.parse` helper. That will give you an array with a string list of the url entries  as there can be multiple uris in the list